docs/csharp/tutorials/exploration/csharp-7.yml

```
@@ -0,0 +1,536 @@
+    > delivered in the framework. Example include `async` and `await`
+    > relying on the `INotifyCompletion` interface, and LINQ relying
+    > on `IEnumerable<T>`. However, the delivery mechanism is changing
+    > as .NET is becoming more platform independent. The .NET Framework
+    > may not always ship on the same cadence as the language compiler. When new language
+    > features rely on new types, those types will be available as NuGet packages when
+    > the language features ship. As these new types get added to the .NET Standard
+    > API and delivered as part of the framework, the NuGet package requirement will
+    > be removed.
+
+    C# provides a rich syntax for classes and structs that is used to explain
+    your design intent. But sometimes that rich syntax requires extra
+    work with minimal benefit. You may often write methods that need a simple
+    structure containing more than one data element. To support these scenarios
+    *tuples* were added to C#. Tuples are lightweight data structures
+    that contain multiple fields to represent the data members.
+    The fields are not validated, and you cannot define your own methods
+
+    > [!NOTE]
+    > Tuples were available before C# 7.0,
+    > but they were inefficient and had no language support.
+    > This meant that tuple elements could only be referenced as
+    > `Item1`, `Item2` and so on. C# 7.0 introduces language support for tuples,
+    > which enables semantic names for the fields of a tuple using new,
+    > more efficient tuple types.
+
+    You can create a tuple by assigning a value to each member:
+
+    [!code-csharp[UnnamedTuple](~/samples/snippets/csharp/new-in-7/program.cs#04_UnnamedTuple "Unnamed tuple")]
+
+    That assignment creates a tuple whose members are `Item1` and `Item2`,
+    which you can use in the same way as <xref:System.Tuple>
+    You can change the syntax to create a tuple that provides semantic
+    names to each of the members of the tuple:
+
+    [!code-csharp[NamedTuple](~/samples/snippets/csharp/new-in-7/program.cs#05_NamedTuple "Named tuple")]
+
+    The `namedLetters` tuple contains fields referred to as `Alpha` and
+    `Beta`. Those names exist only at compile time and are not preserved
+    for example when inspecting the tuple using reflection at runtime.
+
+    In a tuple assignment, you can also specify the names of the fields
+    on the right-hand side of the assignment:
+
+    [!code-csharp[ImplicitNamedTuple](~/samples/snippets/csharp/new-in-7/program.cs#06_ImplicitNamedTuple "Implicitly named tuple")]
+
+    You can specify names for the fields on both the
+    left and right-hand side of the assignment:
+
+    [!code-csharp[NamedTupleConflict](~/samples/snippets/csharp/new-in-7/program.cs#07_NamedTupleConflict "Named tuple conflict")]
+
+    The line above generates a warning, `CS8123`, telling you that the names on the right
+    side of the assignment, `Alpha` and `Beta` are ignored because they conflict
```

```suggestion
    side of the assignment, `Alpha` and `Beta`, are ignored because they conflict
```
Incidentally, running this interactively just generates a "No output" message, not a compiler warning.